bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/FileFormat.java

```diff
@@ -71,20 +90,11 @@ public ImageData[] loadFromStream(LEDataInputStream stream) {
  * return the device independent image array represented by the stream.
  */
 public static ImageData[] load(InputStream is, ImageLoader loader) {
-	FileFormat fileFormat = null;
 	LEDataInputStream stream = new LEDataInputStream(is);
-	for (int i = 1; i < FORMATS.length; i++) {
-		if (FORMATS[i] != null) {
-			try {
-				fileFormat = getFileFormat (stream, FORMATS[i]);
-				if (fileFormat != null) break;
-			} catch (ClassNotFoundException e) {
-				FORMATS[i] = null;
```

Comment:
Can someone explain why some FileFormat implementations could fail to load?
Are there relevant SWT (re-)implementations that have them removed or something like that? Or is this not relevant anymore?